interfaces/sourcegen/sourcegen/_types.py

```diff
@@ -0,0 +1,72 @@
+# This file is part of Cantera. See License.txt in the top-level directory or
+# at https://cantera.org/license.txt for license and copyright information.
+
+from abc import ABCMeta, abstractmethod
+from dataclasses import dataclass, fields
+import os
+
+
+def _unpack(obj):
+    return (getattr(obj, field.name) for field in fields(obj))
+
+
+@dataclass
+class Param:
```

Comment:
```suggestion
class Parameter:
```
Being more descriptive is better, IMO :smile:

Reply:
I definitely agree that longer identifiers should not be feared when they provide additional readability. I think that far too much C[*] code and even the python stuff is too terse. That said, I do think Func and Param are almost universally understood (along with Min, Max, and a scant few others), so I'd like to keep it unless y'all feel strongly the other way.

Reply:
No strong feelings! 😄